app/models/api_key.rb

```diff
@@ -0,0 +1,21 @@
+class ApiKey < ApplicationRecord
+  API_SCOPES = %i[index_rubygems push_rubygem yank_rubygem add_owner remove_owner webhook_actions show_dashboard].freeze
+
+  belongs_to :user
+  validates :user, :name, :hashed_key, presence: true
+  validate :exclusive_show_dashboard_scope, if: :show_dashboard?
```

Comment:
` show_dashboard?`, `index_rubygems?`, `yank_rubygem?` etc as a public API doesn't make much sense to me. `push_rubygem?` for example makes me me think that "we're checking if we should push a rubygem", not "checking if this given API key has permission to push rubygems".

Something like `authorized?(:show_dashboard) or `can_show_dashboard?` to me sounds better and has more clear intent in what we're checking.

Reply:
Good point. I can change this to `can_show_dashboard?` type methods. However, I would like to mention that I am planning to add `pandit` or `cancancan` which would have these abstractions. I didn't want to add these in the same PR because I had crossed 1k lines of changes and ideally, adding these gems would mean we clean up all of our authorization logic.

Let me know if you feel we should use a cleaner name right from start.

Reply:
added `can_..` methods.